.github/workflows/publish.yaml

```diff
@@ -26,37 +36,61 @@ on:
         type: string
 
 jobs:
+  start-runner:
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ${{ vars.AWS_EC2_IMAGE_ID }}
+          ec2-instance-type: ${{ vars.AWS_EC2_INSTANCE_TYPE }}
+          subnet-id: ${{ vars.AWS_DEFAULT_SUBNET_ID }}
+          security-group-id: ${{ vars.AWS_SECURITY_GROUP_ID }}
   get-charm-paths:
     name: Generate the Charm Matrix
-    runs-on: ubuntu-20.04
+    needs: start-runner
+    runs-on: ${{ needs.start-runner.outputs.label }}
     outputs:
       charm_paths_list: ${{ steps.get-charm-paths.outputs.CHARM_PATHS_LIST }}
     steps:
       - uses: actions/checkout@v2
         with: 
           fetch-depth: 0
           ref: ${{ inputs.source_branch }}
+      - name: Install jq
+        run: sudo snap install jq
       - name: Get paths for all charms in repo
         id: get-charm-paths
         run: bash .github/workflows/get-charm-paths.sh
-
-
   publish-charm:
     name: Publish Charm
-    runs-on: ubuntu-20.04
-    needs: get-charm-paths
+    needs: [start-runner, get-charm-paths]
+    runs-on: ${{ needs.start-runner.outputs.label }}
     strategy:
       fail-fast: false
       matrix:
         charm-path: ${{ fromJson(needs.get-charm-paths.outputs.charm_paths_list) }}
-
     steps:
       - name: Checkout
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
           ref: ${{ inputs.source_branch }}
-
+      - name: Setup environment
+        run: apt install -y python3 python-dev python3-dev build-essential libssl-dev libffi-dev libxml2-dev libxslt1-dev zlib1g-dev python3-pip
```

Comment:
Does this work without `sudo`?

Reply:
Runner in this case runs as root thats why the previous run as Ubuntu ... in this case is root